python/ray/serve/_private/controller.py

```diff
@@ -945,13 +948,16 @@ def list_serve_statuses(self) -> List[bytes]:
             statuses.append(self.get_serve_status(name))
         return statuses
 
-    def get_app_config(self, name: str = SERVE_DEFAULT_APP_NAME) -> Optional[Dict]:
+    def get_app_configs(self) -> Optional[Dict[str, ServeApplicationSchema]]:
         checkpoint = self.kv_store.get(CONFIG_CHECKPOINT_KEY)
-        if checkpoint is not None:
-            _, _, _, config_checkpoints_dict = pickle.loads(checkpoint)
-            if name in config_checkpoints_dict:
-                config = config_checkpoints_dict[name]
-                return ServeApplicationSchema.parse_obj(config).dict(exclude_unset=True)
+        if checkpoint is None:
+            return None
+
+        _, _, _, config_checkpoints_dict = pickle.loads(checkpoint)
+        return {
+            app: ServeApplicationSchema.parse_obj(config)
```

Comment:
Why remove `.dict(exclude_unset=True)`?

Reply:
It looks like this previous returned a `Optional[Dict[str, Dict[str, ...]]]`, but the only caller passes it right into a `ApplicationDetails`, which is already expected a `ServeApplicationSchema` https://github.com/ray-project/ray/blob/84379cfe7d6a217848e30d56ffac0bcd5e13c857/python/ray/serve/schema.py#L996 . So it used to go `str -> ServeApplicationSchema.parse_obj -> dict[str, ...] -> ServeApplicationSchema.parse_obj`, which seemed unnecessary.